src/main/kotlin/lotto/domain/selling/ExchangeResult.kt

```diff
@@ -1,16 +1,15 @@
 package lotto.domain.selling
 
-data class ExchangeResult(val details: Map<Int, Int>) {
-    private val totalPrizeMoney = details.toSortedMap().entries.sumBy { findPrizeMoney(it.key) }
+import java.math.BigDecimal
+import java.math.RoundingMode
 
-    fun calculateRateOfReturn(): Float {
-        val origin = details.entries.sumBy { it.value } * Seller.LOTTO_PRICE
-        return totalPrizeMoney.toFloat() / origin
-    }
+data class ExchangeResult(val paymentResult: PaymentResult, val details: Map<Rank, Int>) {
+    private val totalPrizeMoney = details.keys.sumBy { it.prizeMoney }
 
-    companion object {
-        private val PRIZE_MONEY = hashMapOf(3 to 5000, 4 to 50000, 5 to 1_500_000, 6 to 2_000_000_000)
+    val rateOfReturn: BigDecimal =
+        BigDecimal(totalPrizeMoney).divide(BigDecimal(paymentResult.money), EXCHANGE_SCALE, RoundingMode.DOWN)
```

Comment:
`Int`에 대해 호출할 수 있는 `toBigDecimal()`가 있습니다.

```suggestion
        totalPrizeMoney.toBigDecimal().divide(paymentResult.money.toBigDecimal(), EXCHANGE_SCALE, RoundingMode.DOWN)
```